### ISSUE TITLE
Support Focus Visuals

### ISSUE THREAD

**Issue by MartinZikmund** [unoplatform/uno#3093]
## What would you like to be added:

Currently the `UseSystemFocusVisuals` property does not have any effect and for keyboard navigation, the focused visuals are not visible.

## Why is this needed:

Improved keyboard and gamepad navigation support.

## Notes

- Border applies the same CornerRadius as the element
- Border scales and transforms with the element (same RenderTransform)
- Focus visual animates with the element
- Adheres to Viewbox
- Relevant properties
   - `FocusVisualPrimaryBrush` - outer border
   - `FocusVisualPrimaryThickness` - outer border thickness
   - `FocusVisualSecondaryBrush` - inner border
   - `FocusVisualSecondaryThickness` - inner border thickness
   - `FocusVisualMargin` - margin, negative value = visual is "bigger", positive makes it smaller (inside the element)
   - `UseSystemFocusVisuals` - allows turning the default visuals off
- Apply only to `Keyboard` focus, not to `Pointer`
   - In case the control template overrides `FocusState` group, the `Focused` state is applied on keyboard, `PointerFocused` on pointer interaction
   - `UseSystemFocusVisuals`

## To do

- [ ] Move with scroll view
- [ ] Move with animations
- [x] React to `UseSystemFocusVisuals` changes when already focused

## Resources

https://docs.microsoft.com/en-us/windows/uwp/design/input/guidelines-for-visualfeedback

## For which Platform:

- [x] iOS
- [x] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

## Anything else we need to know?

Related: https://github.com/unoplatform/uno/issues/90

**Comment by MartinZikmund** [comment]
Example of focus visual scaled with `ViewBox`:

![image](https://user-images.githubusercontent.com/1075116/114384880-948f9b00-9b8f-11eb-9dc1-503dec328ae4.png)

**Comment by MartinZikmund** [comment]
Initial support is merged, additional features will be added as separate issues
